client/packages.go

```diff
@@ -76,10 +81,94 @@ func (client *Client) Packages() (map[string]Package, error) {
 		return nil, fmt.Errorf("%s: response has no packages", errPrefix)
 	}
 
-	var packages map[string]Package
-	if err := json.Unmarshal(packagesJSON, &packages); err != nil {
+	var packageMap map[string]Package
+	if err := json.Unmarshal(packagesJSON, &packageMap); err != nil {
 		return nil, fmt.Errorf("%s: failed to unmarshal packages: %v", errPrefix, err)
 	}
 
+	var packages Packages
+	for _, pkg := range packageMap {
+		packages = append(packages, pkg)
+	}
+
 	return packages, nil
 }
+
+// IsInstalled returns true if the Package is currently installed
+func (p Package) IsInstalled() bool {
+	return p.Status == StatusInstalled || p.Status == StatusActive
+}
+
+// HasNameContaining returns true if the Package name contains the query. The
```

Comment:
This is switching the topic from API design into testing. The points made are true no matter what your personal choice for testing is, so I hope we can agree about those here. Feek free to get in touch if you'd like to talk about best testing habits, though. Let's just not hold this branch on that.